File: frontend/src/types/prediction.ts
export interface ModelOutput {
  direction: 'up' | 'down'
  probability: number
}

export interface PredictionResponse {
  symbol: string
  prediction: 'up' | 'down'
  confidence: number
  expected_return: number
  xgboost: ModelOutput
  lstm: ModelOutput
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel


class ModelOutput(BaseModel):
    direction: str    # up | down
    probability: float


class PredictionResponse(BaseModel):
    symbol: str
    prediction: str        # up | down
    confidence: float      # 0-1
    expected_return: float  # percentage
    xgboost: ModelOutput
    lstm: ModelOutput
